src/broker.ts

```diff
@@ -51,32 +51,32 @@ export class Broker {
 			} : null
 		};
 		return request(requestOptions)
-			.then((data) => data[0])
+			.then((data: any) => data[0])
 			.then((response) => {
 				if (response.statusCode < 200 && response.statusCode >= 300) {
 					return q.reject(response);
 				}
 				const body = JSON.parse(response.body);
 				return request(_.extend({}, requestOptions, {uri: body._links[`pb:latest-provider-pacts${tag ? "-with-tag" : ""}`].href.replace("{tag}", tag).replace("{provider}", this.options.provider)}));
 			})
-			.then((data) => data[0])
+			.then((data: any) => data[0])
 			.then((response) => response.statusCode < 200 && response.statusCode >= 300 ? q.reject(response) : JSON.parse(response.body));
 	}
 
 	// Find all consumers collates all of the pacts for a given provider (with optional tags)
 	// and removes duplicates (e.g. where multiple tags on the same pact)
 	public findConsumers(): q.Promise<string[]> {
 		logger.debug("Finding consumers");
-		const promises = _.isEmpty(this.options.tags) ? [this.findPacts()] : _.map(this.options.tags, (t) => this.findPacts(t));
+		const promises = _.isEmpty(this.options.tags) ? [this.findPacts()] : _.map(this.options.tags, (t: string) => this.findPacts(t));
 
 		return q.all(promises)
-			.then((values) => _.reduce(values, (array, v) => {
+			.then((values) => _.reduce(values, (array: string[], v: any) => {
 				if (v && v._links && v._links.pacts) {
 					array.push(..._.pluck(v._links.pacts, "href"));
 				}
 				return array;
 			}, []))
-			.catch(() => q.reject(`Unable to find pacts for given provider '${this.options.provider}' and tags '${this.options.tags}'`));
+			.catch(() => q.reject(`Unable to find pacts for given provider '${this.options.provider}' and tags '${this.options.tags}'`) as IWhenable<any>);
```

Comment:
This is an example of that `q` problem - a `q.Promise` is not compatible with a `IWhenable<T>` which it thinks a `q.all(...)` returns. 

Another potential option here is to modify the `q` import to not import the types, but you do lose out on some of the nice code completion and type completeness elsewhere in the file.

Reply:
isn't that the reason why q support generics?